apps/auth/auth0backend.py

```diff
@@ -0,0 +1,47 @@
+from urllib import request
+
+from jose import jwt
+from social_core.backends.oauth import BaseOAuth2
+
+
+class Auth0(BaseOAuth2):
+    """Auth0 OAuth authentication backend"""
+
+    name = "auth0"
+    SCOPE_SEPARATOR = " "
+    ACCESS_TOKEN_METHOD = "POST"
+    REDIRECT_STATE = False
+    EXTRA_DATA = [("picture", "picture"), ("email", "email")]
+
+    def authorization_url(self):
+        return "https://" + self.setting("DOMAIN") + "/authorize"
+
+    def access_token_url(self):
+        return "https://" + self.setting("DOMAIN") + "/oauth/token"
+
+    def get_user_id(self, details, response):
+        """Return current user id."""
+        return details["user_id"]
+
+    def get_user_details(self, response):
+        # Obtain JWT and the keys to validate the signature
+        id_token = response.get("id_token")
+        jwks = request.urlopen(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected.](https://app.codacy.com/gh/Xcov19/covidX/pullRequest?prid=5459956)